File: backend/app/schemas/playlists.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class SelectedTrack(BaseModel):
    title: str
    artistName: str
    uri: str
    isAIRecommended: bool = False
    popularity: Optional[int] = None
    isSelected: bool = False


class PlaylistCreate(BaseModel):
    playlistTitle: str
    selectedTracks: list[SelectedTrack]


class PlaylistAddTracks(BaseModel):
    selectedTracks: list[SelectedTrack]


class PlaylistRead(BaseModel):
    id: int
    title: str
    createdAt: datetime


class PlaylistsPage(BaseModel):
    playlists: list[PlaylistRead]
    total: int

File: frontend/src/models/playlists.ts
export interface SelectedTrack {
  title: string;
  artistName: string;
  uri: string;
  isAIRecommended: boolean;
  popularity: number | null;
  isSelected: boolean;
}

export interface PlaylistCreate {
  playlistTitle: string;
  selectedTracks: SelectedTrack[];
}

export interface Playlist {
  id: number;
  title: string;
  createdAt: string;
}

export interface PlaylistsPage {
  playlists: Playlist[];
  total: number;
}
